plugins/newsql/newsql.c

```diff
@@ -2139,9 +2139,12 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
         Pthread_mutex_unlock(&clnt->wait_mutex);
     }
 
-    if (!query) return NULL;
-    if (errno != 0) {
-        cdb2__query__free_unpacked(query, &pb_alloc);
+    if (!query || (errno != 0)) {
+        logmsg(LOGMSG_ERROR,
+               "%s:%d Error unpacking query error:%s\n",
```

Comment:
I might be a bit nitpicking here: could we add a space between the colon and `%s`?